pkg/tsdb/azuremonitor/types/types.go

```diff
@@ -33,6 +33,11 @@ type AzureMonitorSettings struct {
 	AppInsightsAppId             string `json:"appInsightsAppId"`
 }
 
+// AzureMonitorCustomizedCloudSettings is the extended Azure Monitor settings for customized cloud
+type AzureMonitorCustomizedCloudSettings struct {
+	CustomizedRoutes map[string]AzRoute `json:"customizedRoutes"`
+}
```

Comment:
- The new type added here is to define all settings needed for customized cloud. For now, only the customized routes for Logs, Metrics and ARG are covered in `CustomizedRoutes`. Later, more settings related to authentication will also be added.

- The pre-defined `routes` is not to be modified or replaced by the new property in JSONData, instead a set of routes for Logs, Metrics or ARG in customized cloud is passed to `Routes` of `DatasourceInfo`. I think this can avoid changing the global `routes` instance.

> This brings the question, it seems that you are only interested in the routes for Azure Monitor (not logs or resource graph), but it should be possible to also use those services from an on-premise cloud, isn't it? It would be nice to follow the same structure that well-known clouds.
- I guess the gap comes from the type name contains Azure Monitor. As mentioned, Logs, Metrics and ARG are supposed to be covered in this change.